s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -63,9 +63,12 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
   // def because tokens can expire
   def signingKey = SigningKey(settings.credentialsProvider, CredentialScope(LocalDate.now(), settings.s3Region, "s3"))
 
-  def download(s3Location: S3Location, range: Option[ByteRange] = None): Source[ByteString, Future[ObjectMetadata]] = {
+  def download(s3Location: S3Location,
+               range: Option[ByteRange] = None,
+               sse: Option[ServerSideEncryption] = None): Source[ByteString, Future[ObjectMetadata]] = {
     import mat.executionContext
-    val future = request(s3Location, rangeOption = range)
+    val s3Headers = S3Headers(sse.map(_.headers(GetObject)).getOrElse(Seq.empty))
```

Comment:
a `fold` might be simpler

Reply:
or a `match` which is even more readable IMO (and does not allocate any extra lambdas)

Reply:
LGTM I will changes _map + getOrElse_ by _fold_